### PR TITLE
Introduce a REPL for testing basic CEL expressions.

### PR DESCRIPTION
**Pull request by jnthntatum**
Initial implementation for issue/538. Adds a simple REPL CLI that
supports adding 'let' expressions to introduce variables into the
evaluation context and evaluate cel expressions using the default
environment.

### CODE REVIEW

File: repl/go.mod [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module github.com/google/cel-go/repl
+
+go 1.15
+
+require (
+	github.com/chzyer/readline v1.5.0
+	github.com/google/cel-go v0.11.3
+	google.golang.org/genproto v0.0.0-20220502173005-c8bf987b8c21
+)
```
Comment (TristonianJones) [inline repl/go.mod:9]: You can consider doing a local replace here as well: `replace github.com/google/cel-go => ./..`

This way you're making sure that the code is always building against the parent module.

File: repl/repl.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+				fmt.Printf("Adding let failed:\n%v\n", err)
+			}
+		case "delete":
+			err = eval.DelLetVar(args[0])
+			if err != nil {
+				fmt.Printf("Deleting declaration failed:\n%v\n", err)
+			}
+		case "exit":
+			done = true
+		case "null":
```
Comment (TristonianJones) [inline repl/repl.go]: Any reason why null is special cased here?

File: repl/repl.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+			if err != nil {
+				fmt.Printf("Adding let failed:\n%v\n", err)
+			}
+		case "delete":
+			err = eval.DelLetVar(args[0])
+			if err != nil {
+				fmt.Printf("Deleting declaration failed:\n%v\n", err)
+			}
+		case "exit":
+			done = true
```
Comment (TristonianJones) [inline repl/repl.go]: I think you can replace this with `break`

File: repl/evaluator_test.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+package main
+
+import (
+	"testing"
+)
+
+func TestEvalSimple(t *testing.T) {
+	eval, err := NewEvaluator()
+	if err != nil {
+		panic(err)
```
Comment (TristonianJones) [inline repl/evaluator_test.go]: Prefer `t.Fatalf("NewEvaluator() failed: %v", err)` here and elsewhere

File: repl/evaluator_test.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	_, _, err = eval.Evaluate("[1, 2, 3]")
+
+	if err != nil {
+		t.Errorf("expected ok got: %v", err)
```
Comment (TristonianJones) [inline repl/evaluator_test.go]: For Go test assertions you typically state `got X, wanted Y` as the format, e.g.

```t.Errorf("eval.Evaluate() got %v, wanted non-error", err)```

File: repl/evaluator.go [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+func NewEvaluator() (*Evaluator, error) {
+	var e Evaluator
+	env, err := cel.NewEnv()
+	if err != nil {
+		return nil, err
+	}
+
+	e.env = env
+
+	return &e, nil
```
Comment (TristonianJones) [inline repl/evaluator.go]: Prefer instead writing this as :

```
return &Evaluator{env: env}
```

File: repl/commands_test.go [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+		},
+		{
+			// not yet supported
+			commandLine: `%declare x: int64`,
+		},
+	}
+	for _, tc := range testCases {
+		_, _, _, err := Parse(tc.commandLine)
+		if err == nil {
+			t.Errorf("Expected errr for '%s' but got ok", tc.commandLine)
```
Comment (TristonianJones) [inline repl/commands_test.go]: nit: extra 'r' in 'err', and you'd want to write something closer to `t.Errorf("Parse(%s) failed: %v", tc.commandLine, err)`

**Review by TristonianJones — CHANGES_REQUESTED** [review]

File: repl/go.mod (hunk 1 above)
Comment (jnthntatum) [reply repl/go.mod:9]: done

File: repl/repl.go (hunk 2 above)
Comment (jnthntatum) [reply repl/repl.go]: not really -- folded it into the default case

File: repl/repl.go (hunk 3 above)
Comment (jnthntatum) [reply repl/repl.go]: done

File: repl/evaluator_test.go (hunk 4 above)
Comment (jnthntatum) [reply repl/evaluator_test.go]: done

File: repl/evaluator_test.go (hunk 5 above)
Comment (jnthntatum) [reply repl/evaluator_test.go]: done

File: repl/evaluator.go (hunk 6 above)
Comment (jnthntatum) [reply repl/evaluator.go]: done

File: repl/commands_test.go (hunk 7 above)
Comment (jnthntatum) [reply repl/commands_test.go]: done

**Review by TristonianJones — APPROVED** [review]
In a subsequent PR, would you mind adding a README.md?

**Pull request merged**